lib/rules/no-undef-directives.js

```diff
@@ -0,0 +1,260 @@
+ * @returns { { node: Property, name: string }[] } Array of ASTNodes
+ */
+function getRegisteredDirectives(componentObject) {
+  const directivesNode = componentObject.properties.find(
+    (p) =>
+      p.type === 'Property' &&
+      utils.getStaticPropertyName(p) === 'directives' &&
+      p.value.type === 'ObjectExpression'
+  )
+
+  if (
+    !directivesNode ||
+    directivesNode.type !== 'Property' ||
+    directivesNode.value.type !== 'ObjectExpression'
+  ) {
+    return []
+  }
+
+  return directivesNode.value.properties
+    .filter((node) => node.type === 'Property')
+    .map((node) => {
+      const name = utils.getStaticPropertyName(node)
+      return name ? { node, name } : null
+    })
+    .filter((res) => !!res)
+}
+
+class DefinedInSetupDirectives {
+  constructor() {
+    /**
+     * Directive names
+     * @type {Set<string>}
+     */
+    this.names = new Set()
+  }
+
+  /**
+   * @param {string[]} names
+   */
+  addName(...names) {
+    for (const name of names) {
+      this.names.add(name)
+    }
+  }
+
+  /**
+   * @param {string} rawName
+   */
+  isDefinedDirective(rawName) {
+    const camelName = camelize(rawName)
+    const variableName = `v${casing.capitalize(camelName)}`
+    return this.names.has(variableName)
+  }
+}
+
+class DefinedInOptionDirectives {
+  constructor() {
+    /**
+     * Directive names
+     * @type {Set<string>}
+     */
+    this.names = new Set()
+  }
+
+  /**
+   * @param {string[]} names
+   */
+  addName(...names) {
+    for (const name of names) {
+      this.names.add(name)
+    }
+  }
+
+  /**
+   * @param {string} rawName
+   */
+  isDefinedDirective(rawName) {
+    const camelName = camelize(rawName)
+    if (this.names.has(rawName) || this.names.has(camelName)) {
+      return true
+    }
+
+    // allow case-insensitive ONLY when the directive name itself contains capitalized letters
+    for (const name of this.names) {
+      if (
+        name.toLowerCase() === camelName.toLowerCase() &&
+        name !== name.toLowerCase()
+      ) {
+        return true
+      }
+    }
+
+    return false
+  }
+}
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'disallow use of undefined custom directives',
+      categories: undefined,
+      url: 'https://eslint.vuejs.org/rules/no-undef-directives.html'
+    },
+    fixable: null,
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          ignorePatterns: {
+            type: 'array',
+            items: {
+              type: 'string'
+            },
+            uniqueItems: true
+          }
+        },
+        additionalProperties: true
+      }
+    ],
+    messages: {
+      undef: "The 'v-{{name}}' directive has been used, but not defined."
+    }
+  },
+  /** @param {RuleContext} context */
+  create(context) {
+    const options = context.options[0] || {}
+    /** @type {string[]} */
+    const ignorePatterns = options.ignorePatterns || []
```

Comment:
For consistency, please rename the option to `ignore` and use the `regexp.toRegExpGroupMatcher` util. Then the option supports both literals (`foo` would only ignore `v-foo` but not `v-foo-bar`), and regexp patterns (`/foo/` would also ignore `v-foo-bar`).